components/block/LatestBlocks.tsx

```diff
@@ -152,7 +152,7 @@ const LatestBlocks = (pagination: boolean, displayCard: boolean) => {
 
   const getScreenSize = () => {
```

Comment:
What is the purpose of this function here when you have the screen size checking inside the `MiddleEllipsis` function?

Reply:
This function helps to get the current user's size of the screen to adjust the length of the ellipsed text for xs, sm, md and lg screens

Reply:
I know it's to detect the screen size but feel strange when you have the same function appear in different components. Can this be done in a way that you don't have to repeat the code? Anyway rewrite the whole component as a class first.